.ci/update-tekton-bundle.sh

```diff
@@ -0,0 +1,87 @@
+APPSTUDIO_TASKS_REPO=quay.io/repository/redhat-appstudio-qe/test-images
+PIPELINE_BUNDLE_IMG=quay.io/repository/redhat-appstudio-qe/test-images:pipeline-bundle-$JVM_BUILD_SERVICE_PR_SHA
+BUILD_DEFINITIONS_DIR="$WORKSPACE"/tmp/build-definitions
+
+function cloneBuildDefinitions() {
+    git clone https://$GITHUB_TOKEN@github.com/redhat-appstudio/build-definitions.git "$BUILD_DEFINITIONS_DIR"
+}
+
+function updateAnalyzerImage() {
+    yq e -i "(.[] \
+    | select(.path == \"/spec/steps/-\").value \
+    | select(.name == \"analyse-dependencies-java-sbom\").image) \
+    |= \"$JVM_BUILD_SERVICE_ANALYZER_IMAGE\"" \
```

Comment:
so in debugging https://github.com/openshift/release/pull/30600 I realized last night that the override for the sidecar image at https://github.com/redhat-appstudio/infra-deployments/blob/d156b6cb8ad4180c1ee5d93d68cc45450940996e/hack/preview.sh#L96-L97 is incorrect.

So we now have a bit of a race between this work and https://issues.redhat.com/browse/PLNSRVCE-455

So with 455, we want to remove the use of cluster tasks.  And @stuartwdouglas has already started with the req processor, but that works different, in that the operator just injects the needed content in the taskruns it generates vs. referencing the cluster task.

There ref in the operator to the side car cluster task pulled in via https://github.com/redhat-appstudio/infra-deployments/blob/376acd8c9b4d9dedac1788ba16eee3f76aad7394/components/build/jvm-build-service/kustomization.yaml#L6 over at https://github.com/redhat-appstudio/jvm-build-service/blob/d1b637f56c4df350447685655a373943317c84c2/pkg/reconciler/dependencybuild/dependencybuild.go#L264 as part of build recipe construction.


And we also have references in the bundles to this cluster task and the images used for it, namely https://github.com/redhat-appstudio/build-definitions/blob/main/tasks/patches/add-sbom-and-push.yaml

If we remove the cluster task ref in the operator first, where we again hard code in the task def needed, then we'll want to presumably employ the same env var override pattern @stuartwdouglas is pursuing with https://github.com/redhat-appstudio/infra-deployments/pull/506

I think that is the path we want to prioritize.

But if there are hurdles with that, and we have to continue to live with the `run-maven-component-build` for a while, then we would want to update this PR to take `JVM_BUILD_SERVICE_SIDECAR_IMAGE` and update the ClusterTasks's image as well.

Thoughts @stuartwdouglas @brunoapimentel

Reply:
fyi I think I have the path down for hard coding / in lining the references to https://github.com/redhat-appstudio/jvm-build-service/blob/main/deploy/base/run-maven-component-build-v0.1.yaml and https://github.com/redhat-appstudio/jvm-build-service/blob/main/deploy/base/run-gradle-component-build-v0.1.yaml ... I should have a PR up before my EOB today.

Assuming @stuartwdouglas agrees I am on the right path, we can follow the pattern he did for the req processor and not worry about updating it via this tekton bundle update approach here.

Reply:
won't make EOB today for the https://github.com/redhat-appstudio/jvm-build-service/blob/main/deploy/base/run-gradle-component-build-v0.1.yaml replacement

and @stuartwdouglas reminded me just now in slack their are 2 sidecars and in fact we still have to do the override of the one in add-sbom @brunoapimentel  ... if you can add it to this PR after sorting out the bundle image push, great.  Otherwise, someone can create a follow up PR to build on the base you have here.